harness/determined/tensorboard/util.py

```diff
@@ -26,3 +36,17 @@ def find_tb_files(base_dir: pathlib.Path) -> List[pathlib.Path]:
         return []
 
     return [file for filetype in tb_file_types for file in base_dir.rglob(filetype)]
+
+
+def get_rank_aware_path(p: pathlib.Path, rank: int) -> pathlib.Path:
+    for ext in profiler_file_extensions:
+        if p.match(f"*{ext}"):
+            print(f"matching *{ext}")
+            num_parts = ext.count(".")
+            while num_parts > 0:
+                p = p.with_suffix("")
+                num_parts -= 1
+            p = p.with_name(f"{p.name}#{rank}")
+            p = p.with_suffix(ext)
+            return p
+    return p
```

Comment:
can you explain to me what this is doing?  I don't think I understand what it's trying to accomplish.

Reply:
I added a docstring explaining the purpose. It converts `/tmp/tensorboard-30.ed56557c-000c-4794-aa3e-eaffac608adc.1/plugins/profile/2022_05_13_15_25_41/ip-172-31-8-212.input_pipeline.pb` to `/tmp/tensorboard-30.ed56557c-000c-4794-aa3e-eaffac608adc.1/plugins/profile/2022_05_13_15_25_41/ip-172-31-8-212#7.input_pipeline.pb` for `rank=7`.